### ISSUE TITLE
AREA_SQUARE_METERS was used from tuya_local, this is a deprecated constant which will be removed in HA Core 2025.12

### ISSUE THREAD

**Issue by felipecrs** [make-all/tuya-local#2596]
### Description

Found this in my HA logs:

```
Logger: homeassistant.const
Source: helpers/deprecation.py:222
First occurred: 1:15:14 PM (1 occurrences)
Last logged: 1:15:14 PM

AREA_SQUARE_METERS was used from tuya_local, this is a deprecated constant which will be removed in HA Core 2025.12. Use UnitOfArea.SQUARE_METERS instead, please report it to the author of the 'tuya_local' custom integration
```

### Confirmation

- [X] I am sure this is a bug or improvement, that is well enough described that it can be implemented.
- [ ] If this is for a specific device, the device config is mentioned above, or diagnostics are included.

- [ ] Log messages or diagnostics relevant to the issue are included.

- [X] This is not requesting addition of a new device.

**Comment by make-all** [comment]
This is new in HA 2024.12, which isn't even released yet, so it is not appropriate to change at this time.

**Comment by felipecrs** [comment]
That's totally right, it's not actionable ATM.

**Comment by make-all** [comment]
None of the documentation related to this change has been updated yet, and it is not announced on the Developer blog, but it looks like in addition to replacing the AREA_SQUARE_METERS constant with an enum expanded to cover other units, there is also a new "area" class available for sensor and number entities in 2024.12.

**Comment by smarthomefamilyverrips** [comment]
Logger: homeassistant.const
Source: helpers/deprecation.py:222
First occurred: 11:23:30 (1 occurrences)
Last logged: 11:23:30

AREA_SQUARE_METERS was used from tuya_local, this is a deprecated constant which will be removed in HA Core 2025.12. Use UnitOfArea.SQUARE_METERS instead, please report it to the author of the 'tuya_local' custom integration
